serverless/serverless
# Exclude Node.js dev dependencies in .zip file

## What did you implement:

Closes #2709

Exclude the `dev` node_modules when zipping to decrease the bundle size.

## How did you implement it:

Get a list of the production node_modules. Add the `node_modules/**` glob to the `exclude` and re-include the production `node_modules` via corresponding `<package-name>/**` globs in the `include` glob.

## How can we verify it:

Create a service and do a `npm init`.

Run `npm install --save node-fetch` and `npm install --save-dev jest`.

Run `serverless package` and inspect the zip file. It should only include the dependencies necessary for the `node-fetch` module (and not the ones for `jest`).

**Note:** please use nested directories with dependencies as well. Those should also only include the dev dependencies after packaging.

## Todos:


***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO
